test/Tester/CancellationTests/CancellationTokenPassingTests.cs

```diff
@@ -0,0 +1,161 @@
+            return new TestingSiloHost(new TestingSiloOptions
+            {
+                StartFreshOrleans = false,
+                StartPrimary = true,
+                StartSecondary = true,
+                AdjustConfig = config =>
+                {
+                    config.Globals.DefaultPlacementStrategy = "ActivationCountBasedPlacement";
+                }
+            });
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task GrainTaskCancellation()
+        {
+            foreach (var delay in cancellationDelaysInMS.Select(TimeSpan.FromMilliseconds))
+            {
+                var grain = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+                var tcs = new GrainCancellationTokenSource();
+                var wait = grain.LongWait(tcs.Token, TimeSpan.FromSeconds(10));
+                await Task.Delay(delay);
+                await tcs.Cancel();
+                await Assert.ThrowsAsync<TaskCanceledException>(() => wait);
+            }
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task TokenPassingWithoutCancellation_NoExceptionShouldBeThrown()
+        {
+            var grain = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var tcs = new GrainCancellationTokenSource();
+            await grain.LongWait(tcs.Token, TimeSpan.FromMilliseconds(100));
+            Assert.True(true);
+        }
+
+
+        [Fact, TestCategory("Functional")]
+        public async Task PreCancelledTokenPassing()
+        {
+            var grain = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var tcs = new GrainCancellationTokenSource();
+            await tcs.Cancel();
+            var wait = grain.LongWait(tcs.Token, TimeSpan.FromSeconds(10));
+            await Assert.ThrowsAsync<TaskCanceledException>(() => wait);
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task CancellationTokenCallbacksExecutionContext()
+        {
+            var grain = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var tcs = new GrainCancellationTokenSource();
+            var wait = grain.CancellationTokenCallbackResolve(tcs.Token);
+            await Task.Delay(100);
+            await tcs.Cancel();
+            var result = await wait;
+            Assert.Equal(true, result);
+        }
+
+        [Fact, TestCategory("Functional")]
+        public async Task CancellationTokenCallbacksThrow_ExceptionShouldBePropagated()
+        {
+            var grain = HostedCluster.GrainFactory.GetGrain<ILongRunningTaskGrain<bool>>(Guid.NewGuid());
+            var tcs = new GrainCancellationTokenSource();
+            var wait = grain.CancellationTokenCallbackThrow(tcs.Token);
+            await Task.Delay(100);
+            await Assert.ThrowsAsync<AggregateException>(() => tcs.Cancel());
```

Comment:
`ThrowsAsync` returns the exception. You can assert that the inner exception is of the same type as you actually thrown in your grain (first change the exception type to be something more specialized than just `Exception`).